src/pages/MyProfile.jsx

```diff
@@ -1,15 +1,240 @@
-import React from "react";
+import React, {useEffect, useState} from "react";
+import PopupJS from "reactjs-popup";
 import PageLayout from "../components/PageLayout";
 import Header from "../components/Header";
 import Footer from "../components/Footer";
+import api from "../http/api";
+
+import 'reactjs-popup/dist/index.css';
+import hide from "../assets/hide-pass.svg";
+import show from "../assets/show-pass.svg";
+import {useNavigate, useParams} from "react-router-dom";
+import {toast} from "react-toastify";
 
 function MyProfile() {
+
+    const [profileData, setProfileData] = useState({
+        id: '',
+        username: '',
+        firstName: '',
+        lastName: '',
+        roles: ''
+    });
+
+
+    useEffect(() => {
+        const fetchData = async () => {
+            try {
+                await api.get('/stats/user', {
+                    headers: {
+                        'Authorization': 'Bearer ' + localStorage.getItem('token')
+                    }
+                }).then( (res) => setProfileData(res.data));
+               } catch (error) {}
+        };
+
+        fetchData();
+
+    }, []);
+
+    const sendPasswordReset = async () => {
+        try {
+            await api.post('/auth/reset', {
+                headers: {
+                    'Authorization': 'Bearer ' + localStorage.getItem('token')
+                }
+            });
+        } catch (error) {}
+    };
+
+    const invalidatePatient = async () => {
+        try {
+            await api.delete('/patient/' + profileData.id, {
+                headers: {
+                    'Authorization': 'Bearer ' + localStorage.getItem('token')
+                }
+            });
+        } catch (error) {}
+    };
+
+    const tableRowStyle = {
+        textAlign: 'center',
+        fontSize: 25,
+        padding: '10px',
+        paddingRight: '20px'
+    };
+
+
+    const buttonStyle = {
+        backgroundColor: '#57b0ec',
+        color: '#fff',
+        padding: '8px 12px',
+        border: 'none',
+        cursor: 'pointer',
+        fontSize: 15,
+        fontWeight: '600',
+        borderRadius: '8px'
+    };
+
+    const navigate = useNavigate();
+    const { tkn } = useParams();
+
+    const [showPass, setShowPass] = useState(false);
+    const [showConfirmPass, setShowConfirmPass] = useState(false);
+    const [data, setData] = useState({
+        newPass : '',
+        confirmPass : '',
+        token : tkn
+    });
+
+    const togglePass = () => {
+        setShowPass(prevState => !prevState);
+    }
+
+    const toggleConfirmPass = () => {
+        setShowConfirmPass(prevState => !prevState);
+    }
+    const handleChange = (e) => {
+        const { name, value } = e.target;
+        setData({...data, [name] : value});
+        e.target.focus();
+        console.log("Name: " + name + " , value: " + value);

```

Comment:
Remove logging. Logging should not be happening in production. Specialy logging of sensitive data such as passwords in this case.